{
 "metadata": {
  "name": "",
  "signature": "sha256:bdf3641f0ab7ba41b15be81ca2f8dad7c4a99336acd979c70281098a939cdd86"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Notes\n",
      "\n",
      "- Modify writeScriptHeader\n",
      "  * Load modules\n",
      "- \"main\" starts around line 940\n",
      "  * wrk is defined on 1130\n",
      "  * really starts at 1336 (call runCA)\n",
      "\n",
      "\n",
      "## Running the pipeline\n",
      "\n",
      "### Gatekeeper\n",
      "\n",
      "### Overlapper (MHAP)\n",
      "\n",
      "### Correction\n",
      "  - \n",
      "\n",
      "\n",
      "### Partition\n",
      "  - ~ 500 MB / RAM\n",
      "  - Parallelizable (199 independent jobs)\n",
      "  - 8 cores"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "## past approaches they have tried, and why they failed/what status was\n",
      "PBJelly + Quiver: gap filling\n",
      "MHAP -> low coverage after error correcting\n",
      "\n",
      "## current approach you're trying\n",
      "PBcR-MHAP\n",
      "\n",
      "## what problems you are running into, or that you anticipate running into\n",
      "PBcR-MHAP is based on wgs-assembler (Celera). They only support SGE and LSF.\n",
      "Submitting a single big job consumes too much memory (gatekeeper -> 1.5TB) or could use more CPU (overlapper) on some steps\n",
      "\n",
      "## what other approaches people are using\n",
      "Josh:\n",
      "I  just worked on a fungus genome which we used PacBio (Titus is involved  with this).  I used PBJelly and then SPAdes and got a really good  assembly when paired with our Illumina data from the original sequencing  project (PacBio was for genome scaffolding improvement).\n",
      "\n",
      "Eichler lab:\n",
      "HGAP (using a modified Blasr, but they just added some outputs, no core modifications)\n",
      "\n",
      "Dazzler - http://dazzlerblog.wordpress.com/\n",
      "Currently just the aligner, need to figure out how to use together with other parts of the pipeline.\n",
      "Talk with Jason Chin, he done that for big genomes\n",
      "\n",
      "ectools - https://github.com/jgurtowski/ectools\n",
      "\n",
      "dbg2olc - http://arxiv.org/abs/1410.2801 http://sites.google.com/site/dbg2olc/.\n",
      "claim to be superfast\n",
      "\n",
      "\n",
      "http://wgs-assembler.sourceforge.net/wiki/index.php/RunCA\n",
      "\n",
      "\n",
      "## Adam tips\n",
      "\n",
      "### Hybrid\n",
      "\n",
      "dbg2olc\n",
      "  - arxiv\n",
      "  - u. mariland\n",
      "\n",
      "lordec\n",
      "\n",
      "ec2tools\n",
      "\n",
      "### pure\n",
      "\n",
      "pacbio2ca\n",
      "\n",
      "\n",
      "- Align with reference (Blasr, galGal4)\n",
      "- Smaller subsets:\n",
      "  * Unmapped reads\n",
      "  * Reads mapping partially to micro chromosomes\n"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}